pymatgen/analysis/phase_diagram.py

```diff
@@ -139,21 +115,47 @@ def __init__(self, entry, chempots, name=None):
             name: Optional parameter to name the entry. Defaults to the reduced
                 chemical formula of the original entry.
         """
-        comp = entry.composition
+        super().__init__(
+            entry.composition,
+            entry.energy,
+            name if name else entry.name,
+            entry.attribute if hasattr(entry, "attribute") else None
+        )
+        # NOTE if we init GrandPotPDEntry from ComputedEntry _energy is the
+        # corrected energy of the ComputedEntry hence the need to keep
+        # the original entry to not lose data.
         self.original_entry = entry
-        self.original_comp = comp
-        grandpot = entry.energy - sum([comp[el] * pot for el, pot in chempots.items()])
+        self.original_comp = self._composition
         self.chempots = chempots
-        new_comp_map = {el: comp[el] for el in comp.elements if el not in chempots}
-        super().__init__(new_comp_map, grandpot, entry.name)
-        self.name = name if name else entry.name
 
     @property
-    def is_element(self):
+    def composition(self) -> Composition:
+        """The composition after removing free species
+
+        Returns:
+            Composition
+        """
+        return Composition(
+            {el: self._composition[el] for el in self._composition.elements
+             if el not in self.chempots}
+        )
+
+    @property
+    def chemical_energy(self):
```

Comment:
New property has been added without tests?

Reply:
I think this is tested indirectly by tests on the `energy` property which is now derived from this property and the `_energy`.

Should I add a direct test for coverage?

Reply:
Thank you, a standalone test would be good -- if we have a new, documented, public property, we should make sure to test it returns what we expect explicitly.

Reply:
test added